### PR TITLE
Using obfuscated_id instead of search of person

### LINKED ISSUE #122: App does not return to login screen after token is deleted in TUM Online
When I deleted all my tokens in tum online, the app didnt display the login screen nor displayed me as a Stranger. The app crashed when i tried to click on my name. (#118) The calendar kept working (cache).

### LINKED ISSUE #205: Use TUM id endpoint to receive person details Feature
The endpoint webservicesbasic.id now also returns the id for the current user:
```
<?xml version="1.0" encoding="utf-8"?>
<rowset>
<row>
    <kennung>ga97lod</kennung>
    <vorname>Kordian</vorname>
    <familienname>Bruck</familienname>
    <obfuscated_id>6*D0780F2xxx</obfuscated_id>
    <obfuscated_ids>
        <student>5*F788EDBExxx</student>
        <bedienstete>3*F84C76D1xxx</bedienstete>
        <extern>6*D0780F2xxx</extern>
    </obfuscated_ids>
</row>
</rowset>
```

we can then use the webservicesbasic.personDetails to receive the users image

### PR DESCRIPTION
**Pull request by mathiasquintero**
### Issue
Fixes #122
Fixes #205

### CODE REVIEW

**Comment by kordianbruck** [comment]
This is not correct. The person search for TUMonline should not longer be used to lookup the ID of a student. Instead please use the dedicated API for the token, that will return the current users IDs: https://github.com/TCA-Team/TumCampusApp/blob/master/app/src/main/java/de/tum/in/tumcampusapp/activities/wizard/WizNavCheckTokenActivity.java#L101

**Comment by mathiasquintero** [comment]
@kordianbruck We've been using this request to fetch the minimum required info to display in the more view which is name, id and profile image.

If we shouldn't do this, what should we do to get the image? My other idea would be fetching the contact info (I'll guess the picture is there as well) as well here with the id but that takes very long

**Comment by mathiasquintero** [comment]
@kordianbruck Are you cool with this behavior?

File: TUM Campus App/TUMOnline.swift [hunk 1]
```diff
@@ -39,4 +46,35 @@ struct TUMOnlineAPI: API {
...
+    
+    func onError(call handler: @escaping (Error) -> ()) {
+        queue.async(flags: .barrier) {
+            self.errorHandlers.append(handler)
+        }
+    }
+    
+    func handle(error: Error, from method: HTTPMethod, at endpoint: TUMOnlineEndpoint) {
+        removeCache(for: endpoint)
+        let errorHandlers = queue.sync { self.errorHandlers }
```
[the remark this answers is not in the document]
Comment (mathiasquintero) [reply TUM Campus App/TUMOnline.swift]: That's because it's the same

**Comment by kordianbruck** [comment]
@mathiasquintero I think its good. Of course I don't understand all of it, but looks fine to me.
@TG908 @mammuth thoughts?

**Comment by mammuth** [comment]
Didn't test it but the overall concept seems good! 👍

**Pull request merged**